src/library/searchquery.cpp

```diff
@@ -166,6 +173,38 @@ QString TextFilterNode::toSql() const {
     return concatSqlClauses(searchClauses, "OR");
 }
 
+CrateFilterNode::CrateFilterNode(const QSqlDatabase& database,
+                                 const QString& argument,
+                                 const CrateStorage* pCrateStorage)
+    : m_database(database),
+      m_argument(argument),
+      m_pCrateStorage(pCrateStorage) {
+    init();
+}
+
+void CrateFilterNode::init() {
+    CrateTrackSelectResult crateTracks(
+        m_pCrateStorage->selectTracksSortedByCrateNameLike(m_argument));
+
+    while (crateTracks.next()) {
+        m_trackIds.push_back(crateTracks.trackId());
+    }
+}
+
+bool CrateFilterNode::match(const TrackPointer& pTrack) const {
+    return std::binary_search(m_trackIds.begin(), m_trackIds.end(), pTrack->getId());
+}
+
+QString CrateFilterNode::toSql() const {
```

Comment:
Indeed, right now I run a query in CrateFilterNode::Init() function to fill the vector, and then I return the same SQL to be run by the query in BaseTrackCache::filterAndSort(). Maybe by using lazy evaluation this would work, but right now I run the query 2 times.